WordPress/gutenberg
# Font Library REST API endpoints: address initial feedback from feature branch

## What?

Addresses the initial feedback in https://github.com/WordPress/gutenberg/pull/57688, including

- Registering `wp_font_face` with `show_in_rest` as `true` and removing the manual route registration
- Removing the constructors of both `WP_REST_Font_Families_Controller` and `WP_REST_Font_Faces_Controller`, and depending on the parent constructor
- Removing the `register_routes` method of `WP_REST_Font_Families_Controller` and depending on the parent method
- Replacing custom `permission_callback` methods and overriding the default ones, where needed
- Adding a context param to both `WP_REST_Font_Families_Controller` and `WP_REST_Font_Faces_Controller` that defaults to `edit`
- Adding tests for context param
- Conditionally adding all fields in `prepare_item_for_response` to support the `_fields` arg
- Replace `get_posts` calls with `WP_Query` and turning off the meta and term caches
- Adding translation functions to strings where they were missing
- Unsetting unwanted collection params from `parent::get_collection_params()`

Also, some bug fixes and improvements

- Turning of batching for both `WP_REST_Font_Families_Controller` and `WP_REST_Font_Faces_Controller`, since it doesn't work with file uploads
- Prevent font faces from being created or deleted when the parent font family id in the route is invalid
- Fix race condition that sometimes causes unit tests to fail
- Renaming hooked functions for deleting font family and font face posts to make it clearer what they're for
- Adding a test to make sure embedding works correctly

## Why?

See https://github.com/WordPress/gutenberg/issues/55278

## Testing Instructions

Minimal functionality changes

- Review code changes against feedback comments
- Smoke test font library
- Ensure unit tests are still passing
- Try creating or deleting a font face using an invalid font family id in the url: you should now get an error
